### PR TITLE
[Enhancement] to_bitmap supports int types

### PR DESCRIPTION
**Pull request by LiShuMing**
## Problem Summary:
Fixes # (issue)

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3

### CODE REVIEW

File: be/src/exprs/bitmap_functions.cpp [hunk 1]
```diff
@@ -42,20 +43,35 @@ StatusOr<ColumnPtr> BitmapFunctions::to_bitmap(FunctionContext* context, const s
...
-                                                   "18446744073709551615 currently",
-                                                   slice.to_string())
-                                       .c_str());
+                builder.append_null();
+                continue;
+            }
+        } else {
+            auto raw_value = viewer.value(row);
+            // To be compatible with varchar type, set it null if raw value is less than 0.
+            if (UNLIKELY(raw_value < 0)) {
```
Comment (trueeyu) [inline be/src/exprs/bitmap_functions.cpp:64]: check > uint64::max()

File: gensrc/script/functions.py [hunk 2]
```diff
@@ -548,7 +548,7 @@
     [80041, 'hll_deserialize', 'HLL', ['VARCHAR'], 'HyperloglogFunctions::hll_deserialize'],
 
     # bitmap function
-    [90010, 'to_bitmap', 'BITMAP', ['VARCHAR'], 'BitmapFunctions::to_bitmap', False],
+    [90010, 'to_bitmap', 'BITMAP', ['VARCHAR'], 'BitmapFunctions::to_bitmap<TYPE_VARCHAR>', False],
```
Comment (trueeyu) [inline gensrc/script/functions.py:551]: 90011, 90012?

File: gensrc/script/functions.py [hunk 3]
```diff
@@ -570,6 +570,11 @@
     [91000, 'sub_bitmap', 'BITMAP', ['BITMAP', 'BIGINT', 'BIGINT'], 'BitmapFunctions::sub_bitmap', False],
     [91001, 'bitmap_subset_limit', 'BITMAP', ['BITMAP', 'BIGINT', 'BIGINT'], 'BitmapFunctions::bitmap_subset_limit', False],
     [91002, 'bitmap_subset_in_range', 'BITMAP', ['BITMAP', 'BIGINT', 'BIGINT'], 'BitmapFunctions::bitmap_subset_in_range', False],
+    [91003, 'to_bitmap', 'BITMAP', ['BOOLEAN'], 'BitmapFunctions::to_bitmap<TYPE_BOOLEAN>', False],
+    [91004, 'to_bitmap', 'BITMAP', ['TINYINT'], 'BitmapFunctions::to_bitmap<TYPE_TINYINT>', False],
+    [91005, 'to_bitmap', 'BITMAP', ['SMALLINT'], 'BitmapFunctions::to_bitmap<TYPE_SMALLINT>', False],
+    [91006, 'to_bitmap', 'BITMAP', ['INT'], 'BitmapFunctions::to_bitmap<TYPE_INT>', False],
+    [91007, 'to_bitmap', 'BITMAP', ['BIGINT'], 'BitmapFunctions::to_bitmap<TYPE_BIGINT>', False],
```
Comment (trueeyu) [inline gensrc/script/functions.py:578]: support LARGEINT

**Review by trueeyu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/AggregateFunctionRewriter.java|1|1|100.00%|[]|

**Review by imay — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

File: be/src/exprs/bitmap_functions.cpp [hunk 4]
```diff
@@ -42,20 +43,35 @@ StatusOr<ColumnPtr> BitmapFunctions::to_bitmap(FunctionContext* context, const s
...
         }
 
-        StringParser::ParseResult parse_result = StringParser::PARSE_SUCCESS;
-
-        auto slice = viewer.value(row);
-        auto value = StringParser::string_to_unsigned_int<uint64_t>(slice.data, slice.size, &parse_result);
+        uint64_t value;
+        if constexpr (lt_is_integer<LT> || lt_is_boolean<LT>) {
+            auto raw_value = viewer.value(row);
+            // To be compatible with varchar type, set it null if raw value is less than 0 and less than uint64::max.
```
Comment (silverbullet233) [inline be/src/exprs/bitmap_functions.cpp:49]:
```suggestion
            // To be compatible with varchar type, set it null if raw value is less than 0 or larger than uint64::max.
```

**Comment by silverbullet233** [comment]
add some cases in sql-test?

**Review by satanson — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.3

**Pull request merged**